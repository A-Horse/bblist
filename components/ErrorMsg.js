import React, { Component } from 'react';
import PropTypes from 'prop-types';
import DOM from 'react-dom-factories';

import 'style/component/error-msg.scss';

export class ErrorMsg extends Component {
  static propTypes = {
    messages: PropTypes.array.isRequired
  };

  renderMessages() {
    return (
      <ul>
        {this.props.messages.map((message, i) => {
          return (
            <li key={i}>
              {message}
            </li>
          );
        })}
      </ul>
    );
  }

  render() {
    if (!this.props.messages.length) {
      return DOM.noscript();
    }
    return (
      <div className="error-msg">
        {this.renderMessages()}
      </div>
    );
  }
}
