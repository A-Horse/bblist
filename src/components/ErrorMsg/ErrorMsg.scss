.ErrorMsg {
  color: #ff3535c7;
  font-size: 13px;
  min-height: 18px;
}
