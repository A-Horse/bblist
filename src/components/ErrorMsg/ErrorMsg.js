import React, { Component } from 'react';
import PropTypes from 'prop-types';
import DOM from 'react-dom-factories';
import R from 'ramda';

import './ErrorMsg.scss';

export class ErrorMsg extends Component {
  static propTypes = {
    messages: PropTypes.array
  };

  render() {
    const messages = this.props.messages.filter(R.complement(R.isNil));
    if (!messages.length) {
      return DOM.noscript();
    }
    return (
      <div className="error-msg">
        <ul>
          {messages.map((message, i) => {
            return <li key={i}>{message}</li>;
          })}
        </ul>
      </div>
    );
  }
}

export default ErrorMsg;
