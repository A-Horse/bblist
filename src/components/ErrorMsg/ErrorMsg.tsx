import React from 'react';

import './ErrorMsg.scss';

export function ErrorMsg({ message }) {
  return <div className="ErrorMsg">{message}</div>;
}
