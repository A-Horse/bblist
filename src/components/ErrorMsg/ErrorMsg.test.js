import React from "react";
import ErrorMsg from "./ErrorMsg";
import renderer from "react-test-renderer";

test("ErrorMsg", () => {
  let messages = ["first error", "second error"];
  const tree = renderer.create(<ErrorMsg messages={messages} />);
  expect(tree).toMatchSnapshot();

  messages = [];
  const tree2 = renderer.create(<ErrorMsg messages={messages} />);
  expect(tree2).toMatchSnapshot();

  messages = [null];
  const tree3 = renderer.create(<ErrorMsg messages={messages} />);
  expect(tree3).toMatchSnapshot();
});
