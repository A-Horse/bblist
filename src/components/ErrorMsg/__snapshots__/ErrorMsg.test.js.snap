// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMsg 1`] = `
<div
  className="error-msg"
>
  <ul>
    <li>
      first error
    </li>
    <li>
      second error
    </li>
  </ul>
</div>
`;

exports[`ErrorMsg 2`] = `<noscript />`;

exports[`ErrorMsg 3`] = `<noscript />`;
